supportedBackends/local/src/main/scala/cromwell/backend/impl/local/LocalBackend.scala

```diff
@@ -1,29 +1,227 @@
+  }
+}
+
+class LocalBackend(override val jobDescriptor: BackendJobDescriptor,
+                   override val configurationDescriptor: BackendConfigurationDescriptor) extends BackendJobExecutionActor with SharedFileSystem {
+
+  import LocalBackend._
+  import better.files._
+  import cromwell.core.PathFactory._
+
+  import scala.sys.process._
+
+  // Mutable process variable assigned when the execute method is called.
+  // Needs to be accessible to be killed when the job is aborted.
+  private var process: Option[Process] = None
+
+  val workflowDescriptor = jobDescriptor.descriptor
+  val jobPaths = new JobPaths(workflowDescriptor, backendConfiguration, jobDescriptor.key)
+  val fileSystemsConfig = backendConfiguration.getConfig("filesystems")
+  override val sharedFsConfig = fileSystemsConfig.getConfig("local")
+
+  val call = jobDescriptor.key.call
+  val callEngineFunction = {
+    val callContext = new CallContext(
+      jobPaths.callRoot.toString,
+      jobPaths.stdout.toAbsolutePath.toString,
+      jobPaths.stderr.toAbsolutePath.toString
+    )
+
+    new LocalCallEngineFunctions(fileSystems, callContext)
+  }
+
+  val evaluator = {
+    val declarations = workflowDescriptor.workflowNamespace.workflow.declarations ++ call.task.declarations
+    val unqualifiedWorkflowInputs = workflowDescriptor.inputs map {
+      case (fqn, v) => splitFqn(fqn)._2 -> v
+    }
+    val knownInputs = unqualifiedWorkflowInputs ++ jobDescriptor.symbolMap
+    val lookup = WdlExpression.standardLookupFunction(knownInputs, declarations, callEngineFunction)
+    new ValueEvaluator(lookup, callEngineFunction)
+  }
+
+  private def evaluate(wdlExpression: WdlExpression) = evaluator.evaluate(wdlExpression.ast)
+
+  // TODO determine how this plays with the init/validation work
+  val runtimeAttributes = {
+    val evaluateAttrs = call.task.runtimeAttributes.attrs mapValues evaluate
+    // Fail the call if runtime attributes can't be evaluated
+    val evaluatedAttributes = TryUtils.sequenceMap(evaluateAttrs, "Runtime attributes evaluation").get
+    LocalRuntimeAttributes(evaluatedAttributes)
+  }
+
+  lazy val runsOnDocker = runtimeAttributes.dockerImage.isDefined
+  lazy val processArgs = {
+    val dockerRun = runtimeAttributes.dockerImage map buildDockerRunCommand getOrElse ""
+    Command(Seq("/bin/bash", "-c", s"cat ${jobPaths.script} | $dockerRun /bin/bash <&0"))
+  }
+
+  // Stream Writers
+  lazy val stdoutWriter = jobPaths.stdout.untailed
+  lazy val stderrTailed = jobPaths.stderr.tailed(100)
+
+  override def preStart() = {
+    // workflowPaths.workflowRoot.createDirectories() TODO move to initialize actor
+    jobPaths.callRoot.createDirectories()
+  }
 
   /**
     * Execute a new job.
     */
-  override def execute: Future[BackendJobExecutionResponse] = ???
+  override def execute: Future[BackendJobExecutionResponse] = {
 
-  override protected def jobDescriptor: BackendJobDescriptor = ???
+    val instantiatedCommand = {
```

Comment:
I'm ok with putting it outside, it's just never used anywhere else.
I'd rather make `val`s things that can be though. Even though in this case this will very likely be called only once, since the command can be completely built at instatiation time, making it a val or lazy val makes more sense to me.